module/zfs/spa_errlog.c

```diff
@@ -53,7 +53,17 @@
 #include <sys/spa_impl.h>
 #include <sys/zap.h>
 #include <sys/zio.h>
+#include <sys/dsl_dir.h>
+#include <sys/dmu_objset.h>
+#include <sys/dbuf.h>
 
+#ifdef _KERNEL
```

Comment:
I'm probably missing something obvious, but why is this `#ifdef   _KERNEL`?

Reply:
I am not sure either, it doesn't make much sense so I removed the `#ifdef`.

Reply:
It needs them, otherwise when compiling libzfs it fails with:
```
./../module/zfs/spa_errlog.c:60:12: error: ‘check_filesystem’ declared ‘static’ but never defined [-Werror=unused-function]
 static int check_filesystem(spa_t *spa, uint64_t head_ds,
            ^~~~~~~~~~~~~~~~
../../module/zfs/spa_errlog.c:63:17: error: ‘get_errlog_size’ declared ‘static’ but never defined [-Werror=unused-function]
 static uint64_t get_errlog_size(spa_t *spa, uint64_t spa_err_obj);
                 ^~~~~~~~~~~~~~~
../../module/zfs/spa_errlog.c:64:17: error: ‘get_errlist_size’ declared ‘static’ but never defined [-Werror=unused-function]
 static uint64_t get_errlist_size(spa_t *spa, avl_tree_t *tree);
                 ^~~~~~~~~~~~~~~~
```

Reply:
You might be able to avoid needing these prototypes, if you just move the `check_filesystem()` function to before where it is called in `check_clones()` and `process_error_block` etc

Reply:
Nice suggestion, I moved them around and got rid of the prototypes.